lib/srv/db/postgres/engine.go

```diff
@@ -409,6 +411,55 @@ func (e *Engine) auditUserPermissions(session *common.Session, entries []events.
 	e.Audit.OnPermissionsUpdate(e.Context, session, entries)
 }
 
+// auditResult process backend wire messages and emit result event on
+// appropriate messages.
+func (e *Engine) auditResult(session *common.Session, pgMsg pgproto3.BackendMessage) {
+	var res common.Result
+
+	switch m := pgMsg.(type) {
+	case *pgproto3.CommandComplete:
+		var err error
+		res.AffectedRecords, err = parseCommandResultAffectedRecords(m.CommandTag)
+		if err != nil {
+			e.Log.WithError(err).Debugf("Unable to define the number affected records from command complete")
+		}
+	case *pgproto3.ErrorResponse:
+		res.Error = common.ConvertError(pgconn.ErrorResponseToPgError(m))
+	default:
+		return
+	}
+
+	e.Audit.OnResult(e.Context, session, res)
+}
+
+// parseCommandResultAffectedRecords parses the command complete to retrieve the
+// affected rows information.
+//
+// https://www.postgresql.org/docs/16/protocol-message-formats.html#PROTOCOL-MESSAGE-FORMATS-COMMANDCOMPLETE
+func parseCommandResultAffectedRecords(commandTag []byte) (uint64, error) {
```

Comment:
should we just use `pgconn.CommandTag(commandTag).RowsAffected()`? Not that i don't trust our implementation but using the "official" lib gives me peace of mind so that we don't have to maintain the logic.